src/main/java/org/stellar/sdk/Transaction.java

```diff
@@ -320,10 +323,49 @@ public Builder addTimeBounds(TimeBounds timeBounds) {
       return this;
     }
 
+    /**
+     * Because of the distributed nature of the Stellar network it is possible that the status of your transaction
+     * will be determined after a long time if the network is highly congested.
+     * If you want to be sure to receive the status of the transaction within a given period you should set the
+     * {@link TimeBounds} with <code>maxTime</code> on the transaction (this is what <code>setTimeout</code> does
+     * internally; if there's <code>minTime</code> set but no <code>maxTime</code> it will be added).
+     * Call to <code>Builder.setTimeout</code> is required if Transaction does not have <code>max_time</code> set.
+     * If you don't want to set timeout, use <code>TIMEOUT_INFINITE</code>. In general you should set
+     * <code>TIMEOUT_INFINITE</code> only in smart contracts.
+     * Please note that Horizon may still return <code>504 Gateway Timeout</code> error, even for short timeouts.
+     * In such case you need to resubmit the same transaction again without making any changes to receive a status.
+     * This method is using the machine system time (UTC), make sure it is set correctly.
+     * @param timeout Timeout in seconds.
```

Comment:
Hey @bartekn, what happens when `timeout` is negative? Probably default to `TIMEOUT_INFINITE`?

Reply:
Good catch, fixed!